### PR TITLE
[MOBCORE-4038] Ability to Mock HSS for SIGTRAN load tests

### PR DESCRIPTION
**Pull request by stwang123**
Add ability to start diameter stack that can returns Update Location Answer and Authentication Info Answer

add option to choose which operation to run (ss7 attachment simulation or start diameter stack) when running the jar
`--operation ss7AttachmentSimulation` or `--operation startDiameterStack`

Increase default size for Callbacks map to reduce/eliminate resizing.

Will update the README in https://github.com/eleonoracau/snice-gatling/pull/1

### CODE REVIEW

File: gatling-engine/src/main/resources/gatling.conf [hunk 1]
```diff
@@ -5,6 +5,80 @@
 # This file contains all the settings configurable for Gatling with their default values
 
 gatling {
+
```
Comment (stwang123) [inline gatling-engine/src/main/resources/gatling.conf:8]: Just moving up so I didnt need to scroll all the way down to edit it

File: gatling-engine/src/main/scala/io/snice/gatling/scenarios/Ss7BasicScenarios.scala [hunk 2]
```diff
@@ -16,8 +16,12 @@ object Ss7BasicScenarios {
 
   val feeder = (simulationConfig: Ss7SimulationConfig) => {
     val feederArray = (simulationConfig.imsiStart to simulationConfig.imsiEnd).map {imsi =>
+      val imsiString = imsi.toString
+
+      // converting IMSIs start with 0 to Long  truncates the leading 0s, so this adds leading zeros back if necessary
+      val processedImsi = if (imsiString.length < 15) "0" * (15 - imsiString.length) + imsiString else imsiString
```
Comment (eleonoracau) [inline gatling-engine/src/main/scala/io/snice/gatling/scenarios/Ss7BasicScenarios.scala:22]: we could build a specific IMSI Feeder that returns properly built IMSIs

File: gatling-engine/src/main/scala/io/snice/gatling/simulations/Ss7AttachmentSimulation.scala [hunk 3]
```diff
@@ -29,10 +34,10 @@ class Ss7AttachmentSimulation extends Simulation {
 
   val interval = simConfig.intervalInMinutes.minutes
   val ss7Scenario = Ss7BasicScenarios.ss7Attach(simConfig).inject(
-    atOnceUsers(simConfig.atOnceUsers),
     constantUsersPerSec(simConfig.startConstantUsersPerSec) during interval,
     rampUsersPerSec(simConfig.rampRatePerSec) to simConfig.rampRateTarget during interval,
-    constantUsersPerSec(simConfig.endConstantUsersPerSec).during(interval)
+    nothingFor(3.seconds),
+    constantUsersPerSec(simConfig.endConstantUsersPerSec).during(interval * 2)
```
Comment (eleonoracau) [inline gatling-engine/src/main/scala/io/snice/gatling/simulations/Ss7AttachmentSimulation.scala]: mmh usually we have a config for each step of the scenario. we don't assume it's `interval*2`

File: gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala [hunk 4]
```diff
@@ -125,7 +126,7 @@ class Ss7Client(sgsnStack: MAPStack,
     val additionalVectorsAreForEPS = true
 
     clientDialogMobility.addSendAuthenticationInfoRequest(imsi, numberOfRequestedVectors, segmentationProhibited, immediateResponsePreferred,
-      reSynchronisationInfo, extensionContainer, RequestingNodeType.mmeSgsn, plmnId, numberOfRequestedVectors, additionalVectorsAreForEPS)
+      reSynchronisationInfo, extensionContainer, RequestingNodeType.mmeSgsn, plmnId, 0, additionalVectorsAreForEPS)
```
Comment (eleonoracau) [inline gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala:129]: so you don't get back any vector?

File: gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Engine.scala [hunk 5]
```diff
@@ -53,8 +53,8 @@ class Ss7Engine (config: Ss7EngineConfig) extends StrictLogging {
   def initializeSccpStack(ipChannelType: IpChannelType): SccpStackImpl = {
     val sctpMgmt = initSCTP(ipChannelType)
     val clientM3UAMgmt = initM3UA(sctpMgmt)
-    val sccpStack = initSCCP(clientM3UAMgmt)
     clientM3UAMgmt.startAsp("RASP1")
+    val sccpStack = initSCCP(clientM3UAMgmt)
```
Comment (eleonoracau) [inline gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Engine.scala:57]: from the Jss7 examples the ASP is always started last. you noticed this is better, why?

File: gatling-engine/src/main/scala/io/snice/gatling/Engine.scala [hunk 6]
```diff
@@ -2,21 +2,59 @@ package io.snice.gatling
...
+  override def main(args: Array[String]): Unit = {
+    val usage = """
+      Usage: [--operation operation_name]
+    """
 
-  Option(System.getProperty("simulation")) match {
-    case Some(simulation) => props.simulationClass(simulation)
-    case None => props.simulationClass(classOf[Ss7AttachmentSimulation].getName)
-  }
+    def getOptions(map : Map[String, String], list: List[String]) : Map[String, String] = {
```
Comment (eleonoracau) [inline gatling-engine/src/main/scala/io/snice/gatling/Engine.scala:16]: wondering why we don't make the mocking as part of the Simulation itself. you may have an `Ss7AttachmentSimulationWithMock` that starts also the diameter stack

**Pull request merged**